File: Linguagem C/05_Lista_While/ex_004.c
//4) Faça um programa que leia um inteiro positivo N, calcule e mostre a soma dos N primeiros números
//ímpares. Não deixe o usuário inserir um valor negativo. Se um valor negativo for inserido, continue lendo N.
#include <stdio.h>
int main (){
    int num, i=0,impar=1, soma=0;

    do{
        printf("Digite um numero: ");
        scanf("%d", &num);
    }while(num < 0);
    
    while (i < num){
        soma+=impar;
        impar+=2;
        i++;
    }
    printf("%d", soma);
return 0;
}

File: Python/005_Lista_While/Ex_004.py
"""
4)  Faça  um  programa  que  leia  um  inteiro  positivo  N,  calcule  e  mostre  a  soma  dos  N  primeiros  números 
ímpares. Não deixe o usuário inserir um valor negativo. Se um valor negativo for inserido, continue lendo N.
"""
soma = 0  # Inicializa a variável soma

while True:
    n = int(input("Digite um número para N: "))
    if n > 0:
        break

for num in range(1, 2*n+1, 2):
    soma += num  # Acumula os números ímpares na variável soma

print("A soma dos primeiros", n, "números ímpares é:", soma)
